File: Chapter9 - Developing Apps/Cs/SampleApp_RDotNet_VS2012/IPIDeveloperTech.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SampleApp_RDotNet_VS2012
{
    public interface IPIDeveloperTech
    {
        string GetVersion();
        bool Connect(string PIDataArchiveName = null);
        string[] GetPIDataArchiveNamesList();
        bool ValidateTagNames(string[] piPoints);
        PIValuesList GetRecordedValues(string startTime, string endTime);
        PIValuesList GetInterpolatedValues(string startTime, string endTime, string interval);
        void Disconnect();

        string GetServerName();
    }
}

File: Chapter9 - Developing Apps/Java/SampleApp_JRIandPISystem/src/IPIDeveloperTech.java

public interface IPIDeveloperTech
{
    String GetVersion();
    Boolean Connect(String PIDataArchiveName);
    String[] GetPIDataArchiveNamesList();
    Boolean ValidateTagNames(String[] piPoints);
    PIValuesList GetRecordedValues(String startTime, String endTime);
    PIValuesList GetInterpolatedValues(String startTime, String endTime, String interval);
    void Disconnect();

    String GetServerName();
}